snuba/datasets/events_processor_base.py

```diff
@@ -108,6 +112,32 @@ def extract_promoted_contexts(
     ) -> None:
         raise NotImplementedError
 
+    def __track_latency(
+        self, output: MutableMapping[str, Any], data: MutableMapping[str, Any]
+    ) -> None:
+        # XXX: For now this function is purely for debugging purposes. While
+        # it would be nice to have these kind of statistics for our ingest
+        # system I don't think this hack is the correct solution.
+        now = datetime.utcnow().timestamp()
+        if "timestamp" in output:
+            metrics.timing(
+                "process_timestamp_latency",
+                (now - output["timestamp"].timestamp()) * 1000,
+            )
+        if output.get("received") is not None:
+            metrics.timing(
+                "process_received_latency",
+                (now - output["received"].timestamp()) * 1000,
+            )
+        if data.get("nodestore_insert") is not None:
```

Comment:
Wouldn't it be more straightforward to just track the latency between received time and nodestore_insert time from Sentry and not need pass extra metadata about nodestore (a system Snuba knows nothing about) to Snuba? What additional information does doing it this way provide us?

Reply:
I have already compared nodestore with received, and it's negligible. The question I'm trying to answer now is what is the discrepancy between received, nodestore and the Clickhouse write. It seems like sometimes that lag is on the order of several minutes and I want to see if I can track that down.